File: examples/tutorial/hello/php/func.php
<?php
require 'vendor/autoload.php';

stream_set_blocking(STDIN, 0);
$payload = json_decode(file_get_contents("php://stdin"), true);
if (isset($payload['name'])) {
    echo "Hello ", $payload['name'],"!\n\n";
} else {
    echo "Hello World!\n\n";
}

File: examples/tutorial/hello/python/func.py
import sys
import os
import json

sys.stderr.write("Starting Python Function\n")

name = "World"

try:
  if not os.isatty(sys.stdin.fileno()):
    try:
      obj = json.loads(sys.stdin.read())
      if obj["name"] != "":
        name = obj["name"]
    except ValueError:
      # ignore it
      sys.stderr.write("no input, but that's ok\n")
except:
  pass

print "Hello", name, "!"
